storagemarket/deal_execution.go

```diff
@@ -88,14 +114,32 @@ func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter,
 	// Publish
 	if deal.Checkpoint <= dealcheckpoints.Published {
 		if err := p.publishDeal(ctx, pub, deal); err != nil {
-			return fmt.Errorf("failed to publish deal: %w", err)
+			return &dealMakingError{
+				err: fmt.Errorf("failed to publish deal: %w", err),
+			}
 		}
 	}
+	// Now that the deal has been published, we no longer need to have funds
+	// tagged as being for this deal (the publish message moves collateral
+	// from the storage market actor escrow balance to the locked balance)
+	presp := make(chan struct{}, 1)
```

Comment:
nit: suggest moving this block of code to a function - as a reader of the code I need to check carefully to make sure it doesn't modify any variables that will affect subsequent code

Reply:
done.